src/internal/deploy/assets/assets.go

```diff
@@ -657,26 +662,34 @@ func PachdDeployment(opts *AssetOpts, objectStoreBackend Backend, hostPath strin
 	}
 	envVars = append(envVars, GetSecretEnvVars("")...)
 	envVars = append(envVars, getStorageEnvVars(opts)...)
+
+	name := pachdName
+	command := []string{"/pachd"}
+	if opts.EnterpriseServer {
+		name = enterpriseServerName
+		command = append(command, "--mode=enterprise")
+	}
+
 	return &apps.Deployment{
 		TypeMeta: metav1.TypeMeta{
 			Kind:       "Deployment",
 			APIVersion: "apps/v1",
 		},
-		ObjectMeta: objectMeta(pachdName, labels(pachdName), nil, opts.Namespace),
+		ObjectMeta: objectMeta(name, labels(name), nil, opts.Namespace),
```

Comment:
Give the enterprise server a different deployment name - it should be in a different namespace at least, but this makes it look a little less like another random pachd pod.